cfme/physical/physical_server.py

```diff
@@ -0,0 +1,232 @@
+        """
+        view = navigate_to(self.parent, "All")
+
+        def _looking_for_state_change():
+            entity = view.entities.get_entity(name=self.name)
+            return "currentstate-{}".format(desired_state) in entity.data['state']
+
+        wait_for(_looking_for_state_change, fail_func=view.browser.refresh, num_sec=timeout)
+
+    def get_detail(self, title, field):
+        """Gets details from the details summary tables.
+
+        Args:
+            title (str): Summary Table title
+            field (str): Summary table field name
+
+        Returns: A string representing the entities of the SummaryTable's value.
+        """
+        view = navigate_to(self, "Details")
+        return getattr(view.entities, title.lower().replace(" ", "_")).get_text_of(field)
+
+    @property
+    def exists(self):
+        """Checks if the physical_server exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    def get_db_id(self):
+        if self.db_id is None:
+            self.db_id = self.appliance.physical_server_id(self.name)
+            return self.db_id
+        else:
+            return self.db_id
+
+    def wait_to_appear(self):
+        """Waits for the server to appear in the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the server to appear...")
+        wait_for(
+            lambda: self.exists,
+            message="Wait for the server to appear",
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the server to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the server to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the server to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalServerCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalServer`."""
+
+    ENTITY = PhysicalServer
+
+    def check_physical_servers(self, physical_servers):
+        physical_servers = list(physical_servers)
+        checked_physical_servers = list()
+        view = navigate_to(self, 'All')
+
+        for physical_server in physical_servers:
+            try:
+                view.entities.get_entity(name=physical_server.name, surf_pages=True).check()
+                checked_physical_servers.append(physical_server)
+            except ItemNotFound:
+                raise (ItemNotFound(
+                       'Could not find physical server {} in the UI'.format(physical_server.name)))
```

Comment:
**Question**
Why re-raise the same exception here?  ItemNotFound as raised by `get_entity` contains:
"Entity {keys} isn't found on this page" already, where keys is the passed dictionary of `name=physical_server.name`.  Doesn't appear to add any value to the exception, and the try/except is unnecessary.